download-omnigres-extension.sh

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env bash
+
+set -e
+
+if ! which curl > /dev/null; then
+  echo "curl not found, install curl" && exit 1
+fi
+
+subcommand="$1"
+ext_name="$2"
+ext_version="$3"
+
+OS="$(uname)"
+
+if [ "$OS" = Linux ]; then
+  OS_ARCH="ubuntu-x86-64"
+elif [ "$OS" = Darwin ]; then
+    OS_ARCH="macos-arm"
+else
+  echo "extensions not available for os: $OS" && exit 1
+fi
+
+regex="PostgreSQL[[:space:]]+([0-9]+)\.[0-9]+"
+
+pg_config_version="$(pg_config --version)"
+
+if [[ "$pg_config_version" =~ $regex ]]; then
+    PG_MAJOR_VERSION="${BASH_REMATCH[1]}"
+else
+  echo "pg_config --version output doesn't match the regex: \"$regex\"" && exit 1
+fi
+
+platform_url="https://index.omnigres.com/$PG_MAJOR_VERSION/Release/$OS_ARCH"
+
+tarball_path="/tmp/$ext_name--$ext_version.tar.gz"
+untarred_dir_path="/tmp/$ext_name--$ext_version--untarred"
+
+clean_up () {
+    exit_code=$1
+
+    if [ -f "$tarball_path" ]; then
+      rm "$tarball_path"
+    fi
+
+    if [ -d "$untarred_dir_path" ]; then
+      rm -rf "$untarred_dir_path"
+    fi
+
+    exit $exit_code
+}
+
+fail () {
+  error_msg=$1
+  echo "$error_msg
+
+Usage: $0 <subcommand>
+
+Subcommands:
+  install <ext-name> <ext-version>
+    download and copy the extension files(only if it doesn't exist) to postgres directories of the given omnigres extension version
+
+
+Visit $platform_url/index.json to view available omnigres extension releases"
+
+  clean_up 1
+}
+
+case "$subcommand" in
+
+  install)
+    if [ -z "$ext_name" ] || [ -z "$ext_version" ]; then
+      fail "extension name and version should not be empty"
+    fi
+
+    if ! curl -sSL --fail-with-body "$platform_url/$ext_name/$ext_version?with-dependencies" -o "$tarball_path"; then
+      fail "extension download error: $(cat $tarball_path)"
+    fi
+
+    rm -rf "$untarred_dir_path" && mkdir "$untarred_dir_path" && tar -C "$untarred_dir_path" -xzf "$tarball_path"
+
+    control_and_sql_files_destination="$(pg_config --sharedir)/extension"
+
+    shopt -s nullglob
+    for file in "$untarred_dir_path"/*{.sql,.control}; do
+      if [ ! -f "$control_and_sql_files_destination"/$(basename "$file") ]; then
+        cp -v "$file" "$control_and_sql_files_destination"
+      else
+        echo "skipping copy of $(basename "$file"), already exists"
+      fi
+    done
+    shopt -u nullglob
+
+    if [ -d "$untarred_dir_path/lib" ]; then
+      so_files_destination="$(pg_config --pkglibdir)"
+      shopt -s nullglob
+      for file in "$untarred_dir_path"/lib/*.so; do
+        if [ ! -f "$so_files_destination"/$(basename "$file") ]; then
+          cp -v "$file"  "$so_files_destination"
+        else
+          echo "skipping copy of ./lib/$(basename "$file"), already exists"
```

Comment:
`install` is also very good at this, I think.

Reply:
(this is not a roadblock but an improvement suggestion)

Reply:
went with this approach because `cp -n` errors out in macos if file already exists at the destination,

looking at `install` man page don't see an option for copying the file only if it doesn't exist.

Avoiding overwritting of files while copying because `.so` file of an extension may already be loaded in memory thus want to avoid deleting and creating the same file

Reply:
Got it! It makes sense.

Reply:
I will add the comments

Reply:
Added the comments